### PR TITLE
[Miro] connector improvements

### PR DESCRIPTION
**Pull request by EugeneLightsOn**
### What's being changed:
The Miro connector refactoring and OAuth flow 

### How did you test this change (include any code snippets, API requests, screenshots, or gifs):
image

### CODE REVIEW

File: miro/.env-template [hunk 1]
```diff
@@ -1,2 +1,4 @@
 MIRO_ACCESS_TOKEN=
-MIRO_CONNECTOR_API_KEY=
\ No newline at end of file
+MIRO_SEARCH_LIMIT=20
+MIRO_CONNECTOR_API_KEY=
+MIRO_FIELDS_MAPPING=
```
Comment (daniel-cohere) [inline miro/.env-template:4]: think a newline is missing here

File: miro/README.md [hunk 2]
```diff
@@ -10,9 +10,67 @@ The Miro connector currently allows full-text search of a board's title and desc
 
 To configure this connector you will have to either create a new Miro app, see [here](https://developers.miro.com/docs/rest-api-build-your-first-hello-world-app#step-1-create-your-app-in-miro), or configuring an existing app.
 
-Your app will require at the least the `board:read` permission to allow searching your boards, once that is done you can save and install the app, a message should notify you that the app was successfully installed, and give you an access token. Use this token for the `MIRO_ACCESS_TOKEN` environment variable.
+Your app will require at the least the `board:read` permission to allow searching your boards, once that is done you can save and install the app,
+a message should notify you that the app was successfully installed, and give you an access token. Use this token for the `MIRO_ACCESS_TOKEN` environment variable.
```
Comment (daniel-cohere) [inline miro/README.md:14]:
```suggestion
Your app will require at the least the `board:read` permission to allow searching your boards, once that is done you can save and install the app.
A message should notify you that the app was successfully installed, and give you an access token. Use this token for the `MIRO_ACCESS_TOKEN` environment variable.
```

File: miro/README.md [hunk 3]
```diff
@@ -10,9 +10,67 @@ The Miro connector currently allows full-text search of a board's title and desc
...
+The `MIRO_ACCESS_TOKEN` environment variable must be set if you didn't select the "Expire user authorization token" checkbox during app creation.
+If you selected that checkbox, you will need to use the Oauth flow.
 
-Finally, to protect this connector from abuse, the `MIRO_CONNECTOR_API_KEY` environment variable must be set to a secure value that will be used for this connector's own bearer token authentication.
+### Oauth flow
```
Comment (daniel-cohere) [inline miro/README.md:18]:
```suggestion
### OAuth flow
```

File: miro/README.md [hunk 4]
```diff
@@ -10,9 +10,67 @@ The Miro connector currently allows full-text search of a board's title and desc
...
+The OAuth flow should occur outside of the Connector and Cohere's API will forward the user's access token
```
Comment (daniel-cohere) [inline miro/README.md:19]:
```suggestion
The OAuth flow should occur outside of the Connector by calling Cohere's API. Once authorized, Cohere's API will forward the user's access token
```

File: miro/README.md [hunk 5]
```diff
@@ -10,9 +10,67 @@ The Miro connector currently allows full-text search of a board's title and desc
...
+  "oauth": {
+    "client_id": "{Your App CLIENT-ID}",
+    "client_secret": "{Your App CLIENT-SECRET}",
+    "authorize_url": "https://miro.com/oauth/authorize",
+    "token_url": "https://api.miro.com/v1/oauth/token",
+  }
+}'
+```
+No more configuration is needed here and after successfully registration, 
+Cohere will take care of OAuth steps including passing in the correct headers to your connector.
```
Comment (daniel-cohere) [inline miro/README.md:46]: We might want to mention that users will have to complete the authorization flow here (if they're not using coral/playground) by navigating to the url returned at `/connectors/{id}/oauth/authorize`

**Review by daniel-cohere — APPROVED** [review]
nice!

File: miro/provider/client.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            headers=self.headers,
+            params=params,
+        )
+        if response.status_code != 200:
+            message = response.text or f"Error: HTTP {response.status_code}"
+            raise UpstreamProviderError(message)
+
+        results = response.json()["data"]
+
+        return results
```
Comment (tianjing-li) [inline miro/provider/client.py:35]: just return `response.json()["data"]` directly

File: miro/provider/client.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def get_client():
+    access_token = app.config.get("ACCESS_TOKEN", None)
```
Comment (tianjing-li) [inline miro/provider/client.py:39]: You can save lines 41-42 by changing this to `app.config.get("ACCESS_TOKEN", get_access_token())`

**Review by tianjing-li — APPROVED** [review]
Minor comments

File: miro/.env-template (hunk 1 above)
Comment (EugeneLightsOn) [reply miro/.env-template:4]: Fixed, thanks

File: miro/README.md (hunk 2 above)
Comment (EugeneLightsOn) [reply miro/README.md:14]: Thank you

File: miro/README.md (hunk 3 above)
Comment (EugeneLightsOn) [reply miro/README.md:18]: Thanks

File: miro/README.md (hunk 5 above)
Comment (EugeneLightsOn) [reply miro/README.md:46]: Thanks

File: miro/provider/client.py (hunk 6 above)
Comment (EugeneLightsOn) [reply miro/provider/client.py]: Thanks

File: miro/provider/client.py (hunk 7 above)
Comment (EugeneLightsOn) [reply miro/provider/client.py]: Good point

File: miro/README.md (hunk 4 above)
Comment (EugeneLightsOn) [reply miro/README.md]: Thanks

**Pull request merged**